engine/orchestration/src/com/cloud/agent/manager/ClusteredAgentManagerImpl.java

```diff
@@ -495,28 +495,29 @@ public SocketChannel connectToPeer(final String peerName, final SocketChannel pr
                 }
                 final String ip = ms.getServiceIP();
                 InetAddress addr;
+                int port = Port.value();
                 try {
                     addr = InetAddress.getByName(ip);
                 } catch (final UnknownHostException e) {
                     throw new CloudRuntimeException("Unable to resolve " + ip);
                 }
                 SocketChannel ch1 = null;
                 try {
-                    ch1 = SocketChannel.open(new InetSocketAddress(addr, Port.value()));
+                    ch1 = SocketChannel.open(new InetSocketAddress(addr, port));
```

Comment:
we can however, I avoided much changes, as we were doing something with handshake etc.